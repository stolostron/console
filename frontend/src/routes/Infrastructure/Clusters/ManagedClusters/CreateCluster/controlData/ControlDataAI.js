/* Copyright Contributors to the Open Cluster Management project */

import DetailsForm from '../components/assisted-installer/DetailsForm'
import { automationControlData, appendKlusterletAddonConfig, insertToggleModalFunction } from './ControlDataHelpers'
import { CreateCredentialModal } from '../../../../../../components/CreateCredentialModal'

export const getControlDataAI = (handleModalToggle, includeKlusterletAddonConfig = true) => {
    const controlData = [...controlDataAI]
    appendKlusterletAddonConfig(includeKlusterletAddonConfig, controlData)
    insertToggleModalFunction(handleModalToggle, controlData)
    return controlData
}

export const controlDataAI = [
    ////////////////////////////////////////////////////////////////////////////////////
    ///////////////////////  AI form  /////////////////////////////////////
    {
        id: 'aiDetailStep',
        type: 'step',
        title: 'Cluster details',
    },
    {
        id: 'infrastructure',
        name: 'Infrastructure',
        active: 'Host inventory',
        type: 'reviewinfo',
    },
    {
        id: 'controlplane',
        name: 'Control plane type',
        active: 'Standalone',
        type: 'reviewinfo',
    },
    /////////////////////// ACM Credentials  /////////////////////////////////////
    {
        name: 'creation.ocp.cloud.connection',
        tooltip: 'tooltip.creation.ocp.cloud.connection',
        id: 'connection',
        type: 'singleselect',
        placeholder: 'creation.ocp.cloud.select.connection',
        providerId: ['hybrid', 'hostinventory'],
        validation: {
            notification: 'creation.ocp.cluster.must.select.connection',
            required: false,
        },
        available: [],
        footer: <CreateCredentialModal />,
    },
    {
        id: 'ai',
        type: 'custom',
        component: <DetailsForm />,
        providerId: 'ai',
        mustValidate: true,
        encodeValues: ['pullSecret'],
        additionalProps: {
            promptSshPublicKey: false,
        },
    },
    ...automationControlData,
    {
        id: 'reviewSave',
        type: 'review',
        title: 'Review and save',
        nextButtonLabel: 'Save',
        comment:
            'Ensure these settings are correct. The saved cluster draft will be used to determine the available network resources. Therefore after you press Save you will not be able to change these cluster settings.',
        disableEditorOnSuccess: true,
        disablePreviousControlsOnSuccess: true,
    },
    {
        id: 'aiHostsStep',
        type: 'step',
        title: 'Cluster hosts',
        disabled: true,
    },
    {
        id: 'aiNetworkStep',
        type: 'step',
        title: 'Networking',
        disabled: true,
    },
]

const aiStep = getControlDataAI().find((data) => data.id === 'ai')
aiStep.additionalProps.promptSshPublicKey = true

export default getControlDataAI
