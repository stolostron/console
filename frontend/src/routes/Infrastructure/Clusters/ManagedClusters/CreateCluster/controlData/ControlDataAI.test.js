// Copyright (c) 2022 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import { getControlDataAI } from './ControlDataAI'
import i18next from 'i18next'

const t = i18next.t.bind(i18next)
const handleModalToggle = jest.fn()

describe('Cluster creation control data for AI', () => {
  it('generates correctly', () => {
    expect(getControlDataAI(t, handleModalToggle, true)).toMatchSnapshot()
  })
  it('generates correctly for MCE', () => {
    expect(getControlDataAI(t, handleModalToggle, false)).toMatchSnapshot()
  })
})
