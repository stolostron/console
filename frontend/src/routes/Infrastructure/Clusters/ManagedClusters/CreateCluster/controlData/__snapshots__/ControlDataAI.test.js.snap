// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cluster creation control data for AI generates correctly 1`] = `
Array [
  Object {
    "id": "aiDetailStep",
    "title": "Cluster details",
    "type": "step",
  },
  Object {
    "active": "Host inventory",
    "id": "infrastructure",
    "name": "Infrastructure",
    "type": "reviewinfo",
  },
  Object {
    "active": "Standalone",
    "id": "controlplane",
    "name": "Control plane type",
    "type": "reviewinfo",
  },
  Object {
    "available": Array [],
    "footer": <CreateCredentialModal
      handleModalToggle={[MockFunction]}
    />,
    "id": "connection",
    "name": "Infrastructure provider credential",
    "placeholder": "Select a credential",
    "providerId": Array [
      "hybrid",
      "hostinventory",
    ],
    "tooltip": "The settings that are required for the selected provider. You can select an existing connection, or add a new connection. Cannot be changed after creation.",
    "type": "singleselect",
    "validation": Object {
      "notification": "Select a connection",
      "required": false,
    },
  },
  Object {
    "additionalProps": Object {
      "promptSshPublicKey": true,
    },
    "component": <DetailsForm />,
    "encodeValues": Array [
      "pullSecret",
    ],
    "id": "ai",
    "mustValidate": true,
    "providerId": "ai",
    "type": "custom",
  },
  Object {
    "id": "automationStep",
    "title": "Automation",
    "type": "step",
  },
  Object {
    "id": "chooseTemplate",
    "info": "Optional: Choose an automation template to run Ansible jobs at different stages of the cluster lifecycle.",
    "type": "title",
  },
  Object {
    "component": <AutomationProviderHint
      className="creation-view-controls-hint"
      component="hint"
    />,
    "id": "automationProviderHint",
    "type": "custom",
  },
  Object {
    "id": "templateName",
    "name": "Automation template",
    "onSelect": [Function],
    "placeholder": "Select an automation template",
    "prompts": Object {
      "icon": <ExternalLinkAltIcon
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />,
      "id": "add-automation-template",
      "positionBottomRight": true,
      "prompt": "Add automation template",
      "type": "link",
      "url": "/multicloud/infrastructure/automations/add",
    },
    "tooltip": "Only templates with valid Ansible Automation Platform credentials are listed.",
    "type": "singleselect",
    "validation": Object {
      "required": false,
    },
  },
  Object {
    "component": <TemplateLinkOutControl />,
    "id": "curatorLinkOut",
    "type": "custom",
  },
  Object {
    "component": <TemplateSummaryControl />,
    "id": "curatorSummary",
    "type": "custom",
  },
  Object {
    "active": "",
    "id": "clusterCuratorSpec",
    "reverse": [Function],
    "type": "hidden",
  },
  Object {
    "active": Array [],
    "hidden": true,
    "id": "supportedCurations",
    "type": "values",
  },
  Object {
    "active": "",
    "id": "toweraccess-install",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "toweraccess-upgrade",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "toweraccess-scale",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "toweraccess-destroy",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "reconcilePause",
    "type": "hidden",
  },
  Object {
    "comment": "Ensure these settings are correct. The saved cluster draft will be used to determine the available network resources. Therefore after you press Save you will not be able to change these cluster settings.",
    "disableEditorOnSuccess": true,
    "disablePreviousControlsOnSuccess": true,
    "id": "reviewSave",
    "nextButtonLabel": "Save",
    "title": "Review and save",
    "type": "review",
  },
  Object {
    "disabled": true,
    "id": "aiHostsStep",
    "title": "Hosts",
    "type": "step",
  },
  Object {
    "disabled": true,
    "id": "aiNetworkStep",
    "title": "Networking",
    "type": "step",
  },
  Object {
    "active": true,
    "id": "includeKlusterletAddonConfig",
    "type": "hidden",
  },
]
`;

exports[`Cluster creation control data for AI generates correctly for MCE 1`] = `
Array [
  Object {
    "id": "aiDetailStep",
    "title": "Cluster details",
    "type": "step",
  },
  Object {
    "active": "Host inventory",
    "id": "infrastructure",
    "name": "Infrastructure",
    "type": "reviewinfo",
  },
  Object {
    "active": "Standalone",
    "id": "controlplane",
    "name": "Control plane type",
    "type": "reviewinfo",
  },
  Object {
    "available": Array [],
    "footer": <CreateCredentialModal
      handleModalToggle={[MockFunction]}
    />,
    "id": "connection",
    "name": "Infrastructure provider credential",
    "placeholder": "Select a credential",
    "providerId": Array [
      "hybrid",
      "hostinventory",
    ],
    "tooltip": "The settings that are required for the selected provider. You can select an existing connection, or add a new connection. Cannot be changed after creation.",
    "type": "singleselect",
    "validation": Object {
      "notification": "Select a connection",
      "required": false,
    },
  },
  Object {
    "additionalProps": Object {
      "promptSshPublicKey": true,
    },
    "component": <DetailsForm />,
    "encodeValues": Array [
      "pullSecret",
    ],
    "id": "ai",
    "mustValidate": true,
    "providerId": "ai",
    "type": "custom",
  },
  Object {
    "id": "automationStep",
    "title": "Automation",
    "type": "step",
  },
  Object {
    "id": "chooseTemplate",
    "info": "Optional: Choose an automation template to run Ansible jobs at different stages of the cluster lifecycle.",
    "type": "title",
  },
  Object {
    "component": <AutomationProviderHint
      className="creation-view-controls-hint"
      component="hint"
    />,
    "id": "automationProviderHint",
    "type": "custom",
  },
  Object {
    "id": "templateName",
    "name": "Automation template",
    "onSelect": [Function],
    "placeholder": "Select an automation template",
    "prompts": Object {
      "icon": <ExternalLinkAltIcon
        color="currentColor"
        noVerticalAlign={false}
        size="sm"
      />,
      "id": "add-automation-template",
      "positionBottomRight": true,
      "prompt": "Add automation template",
      "type": "link",
      "url": "/multicloud/infrastructure/automations/add",
    },
    "tooltip": "Only templates with valid Ansible Automation Platform credentials are listed.",
    "type": "singleselect",
    "validation": Object {
      "required": false,
    },
  },
  Object {
    "component": <TemplateLinkOutControl />,
    "id": "curatorLinkOut",
    "type": "custom",
  },
  Object {
    "component": <TemplateSummaryControl />,
    "id": "curatorSummary",
    "type": "custom",
  },
  Object {
    "active": "",
    "id": "clusterCuratorSpec",
    "reverse": [Function],
    "type": "hidden",
  },
  Object {
    "active": Array [],
    "hidden": true,
    "id": "supportedCurations",
    "type": "values",
  },
  Object {
    "active": "",
    "id": "toweraccess-install",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "toweraccess-upgrade",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "toweraccess-scale",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "toweraccess-destroy",
    "type": "hidden",
  },
  Object {
    "active": "",
    "id": "reconcilePause",
    "type": "hidden",
  },
  Object {
    "comment": "Ensure these settings are correct. The saved cluster draft will be used to determine the available network resources. Therefore after you press Save you will not be able to change these cluster settings.",
    "disableEditorOnSuccess": true,
    "disablePreviousControlsOnSuccess": true,
    "id": "reviewSave",
    "nextButtonLabel": "Save",
    "title": "Review and save",
    "type": "review",
  },
  Object {
    "disabled": true,
    "id": "aiHostsStep",
    "title": "Hosts",
    "type": "step",
  },
  Object {
    "disabled": true,
    "id": "aiNetworkStep",
    "title": "Networking",
    "type": "step",
  },
  Object {
    "active": false,
    "id": "includeKlusterletAddonConfig",
    "type": "hidden",
  },
]
`;
